lib/analytics.js

```diff
@@ -0,0 +1,405 @@
+    _log('Analytics on hold until ' + d);
+}
+
+function removeAnalyticsHold () {
+    ON_HOLD_UNTIL = null;
+}
+
+/**
+ * Check if this install has a cordova wrapper with
+ * connectivity awareness. If it does, listen for
+ * network events
+ */
+if (network.connection) {
+    _log('Cordova app with network awareness');
+
+    network.on('online', function () {
+        removeAnalyticsHold();
+        processNext();
+    });
+
+    network.on('offline', function () {
+        setAnalyticsOnHold(30);
+    });
+}
+
+
+/**
+ * Functions for building the requests, and sending them to GA
+ */
+
+var commonAppValues = {};
+function getCommonAppValues () {
+    if (commonAppValues.v) {
+        return commonAppValues;
+    }
+
+    commonAppValues = {
+        // Required
+        v: 1,                               // MP Version
+        tid : config.GA_TRACKING_ID,        // Tracking ID / Property ID.
+        cid : model.data.session.guestId,   // Anonymous Client ID.
+
+        // Additional values we want to use for App Specific measurement
+        an : packageJSON.name,              // Application Name
+        //aid : '',                         // Application ID
+        av : packageJSON.version,           // Application Version
+        //aiid : '',                        // Application Installer ID
+        ds: 'app',                          // Data Source
+        ul: model.locale.toLowerCase(),     // User Language
+        ua: navigator.userAgent             // User Agent
+    };
+    return commonAppValues;
+}
+
+function checkIfHitWasOffline (hit) {
+    if (hit.originalRequestTime) {
+        // we're retrying an earlier request that failed
+        // The value Queue Time (qt) represents the time delta (in milliseconds)
+        // between when the hit being reported occurred and the time the hit was
+        // sent.
+        var now = new Date();
+        var qt = now.getTime() - hit.originalRequestTime.getTime();
+        hit.qt = qt;
+        // use a custom dimension to note this was recorded while offline
+        hit.cd1 = 'offline';
+        return true;
+    } else {
+        hit.cd1 = 'online';
+        return false;
+    }
+}
+
+function attemptSendToGA (hit, callback) {
+    var wasOffline = checkIfHitWasOffline(hit);
+    var gaObj = utils.simpleObjectMerge(getCommonAppValues(), hit);
+    var connectionText = (wasOffline) ? 'Offline' : 'Live';
+
+    if (wasOffline) {
+        // this value doens't need to go to GA
+        delete gaObj.originalRequestTime;
+    }
+
+    // Convert obj to a string for POST request
+    var gaBody = utils.toParameterString(gaObj);
+
+    // Make a POST request to GA
+    xhr({
+        body: gaBody,
+        uri: "https://ssl.google-analytics.com/collect",
+        method: "POST"
+    }, function (err, resp, body) {
+        if (resp.statusCode === 200) {
+            // we're got as far as GA (and must be online)
+            _log('sent hit to GA', connectionText);
```

Comment:
Good call. This is now off by default, except for warnings. It can be enabled with `ANALYTICS_CONSOLE_LOGGING`